lib/obsolete.gd

```diff
@@ -479,6 +479,30 @@ DeclareObsoleteSynonym( "MutableIdentityMat", "IdentityMat", "4.8" );
 ##
 DeclareObsoleteSynonym( "MutableNullMat", "NullMat", "4.8" );
 
+#############################################################################
+##
+#F  IsSemilatticeAsSemigroup( <S> ) is the semigroup <S> a semilattice
+##
+##  <#GAPDoc Label="IsSemilatticeAsSemigroup">
+##  <ManSection>
+##  <Prop Name="IsSemilatticeAsSemigroup" Arg='S'/>
+##
+##  <Description>
+##    <C>IsSemilatticeAsSemigroup</C> returns <K>true</K> if the semigroup
+##    <A>S</A> is a semilattice and <K>false</K> if it is not. <P/>
+##
+##    A semigroup is a <E>semilattice</E> if it is commutative and every
+##    element is an idempotent. The idempotents of an inverse semigroup form a
+##    semilattice.
+##
+##    This is identical to <Ref Prop="IsSemilattice" BookName = "Semigroups"/> #
+##    and is present in &GAP;&nbsp;4.8 #  only for the sake of compatibility with
+##    beta-releases.  #  It should <E>not</E> be used in new code.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareObsoleteSynonym( "IsSemilatticeAsSemigroup", "IsSemilattice", "4.8" );
```

Comment:
Don't use `DeclareObsoleteSynonym` with a property -- it will leave existing code that uses `IsSemilatticeAsSemigroup` broken. Instead, use `DeclareSynonymAttr` or `DeclareObsoleteSynonymAttr`. The former is the one with the best backwards compatibility. The latter will still break code that installs methods using `IsSemilatticeAsSemigroup` as a filter. So unless you are pretty sure there is little to no existing code out there which uses `IsSemilatticeAsSemigroup` as a method filter, the safest choice is `DeclareSynonymAttr`. (The only advantage of `DeclareObsoleteSynonymAttr` is that it prints a warning to users that the name `IsSemilatticeAsSemigroup` is obsolete.)